### DISCUSSION TITLE
Suggestion for control of buildings via sidebar

### DISCUSSION THREAD

**Discussion by BazzaCuda** [Keriew/augustus#562] [Ideas]
(I'm on Windows 10)

Right-clicking on a building (e.g. Warehouse or Granary, etc.) pops up a window with info and buttons to click.  But this interrupts the entire operation of the city until the window is exited. I suggest being able to also Left-click on a building which would show all the info and buttons in the lower part of the sidebar, temporarily covering the standard info (Game Speed, Employment, etc.) that is currently shown there. This lower panel would have all the exact same info as the popup window and all the same buttons, with a standard Exit Window button in the bottom right. The big difference is that having the info and operations available on the sidebar would not interrupt the city, just as the current display of info there doesn't. This would apply to every building type that can be right-clicked; i.e. all of them! :D This way you can examine buildings and make quick changes to the operations of buildings, goods, etc. while your city continues to operate.

Am I correct to assume that on mobile devices you long-press to display the info/operations window? In which case, a short press could be used to show the info/operations in the lower part of the sidebar (?)

What do you think, Keriew? (and others?)

**Comment by BazzaCuda** [comment]
Further to this, how about having a user interface option for all alerts (Emperor requests, invasions, price increases, etc.) to slide out over the sidebar as non-interrupting panels? The only way a game ever stops would be when the user deliberately pauses it.

**Comment by Ouaz** [comment]
> I suggest being able to also Left-click on a building which would show all the info and buttons in the lower part of the sidebar, temporarily covering the standard info

Seems complicated, at least regarding UI design.
For example, I can't imagine the market/warehouse special orders fitting well in the sidebar, even as an option.
Moreover, the extra info panel is dependant of the screen resolution. Roughly below 1600x900, the extra info panel is cut (at 1366x768, it shows only Game Speed and Unemployment stats)

> Further to this, how about having a user interface option for all alerts

I think it was planned... a long time ago. :p A screen to summarize all special orders or something like that.

**Reply by BazzaCuda** [reply]
Ah, ok; fair enough. I'll have a rethink. What is it that makes, say, the Warehouse window what I would call "modal" (Windows dev parlance) and stops the game from running behind it while the window is active? (in contrast to the sidebar which co-functions with the city still running?)
